coalib/bearlib/aspects/meta.py

```diff
@@ -76,21 +77,41 @@ def __repr__(cls):
         return '<%s %s>' % (type(cls).__name__, repr(cls.__qualname__))
 
 
+def isaspect(item):
+    """
+    This function checks whether or not an object is an instance of
+    ``aspectclass``
+    """
+    if not isinstance(item, (aspectclass, aspectbase)):
+        return False
+
+    return True
+
+
+def assert_aspect(item):
+    """
+    This function raises ``aspectTypeError`` when an object is not an instance
+    of ``aspectclass``
+    """
+    class aspectTypeError(TypeError):
```

Comment:
@Techievena @adhikasp The main problem here is that when the `aspectTypeError` class is defined inside the function, a new class object will be created on every function call. In Python, such inner definitions are not static like in compiled languages that allow such things:

```python
def create_class():
    class IWasCreatedInsideAFunction:
        pass

    return IWasCreatedInsideAFunction
```

```python
>>> create_class() is create_class()
False
```

The class must leave its cozy function scope and move to the harsh global module level ;)